lib/colorls/core.rb

```diff
@@ -217,13 +218,36 @@ def mtime_info(stat)
       mtime
     end
 
+    def git_info(path,content)
+      until File.exist?('.git') # check whether the repository is git controlled
+        return '' if Dir.pwd=='/'
+        Dir.chdir('..')
+      end
+      instance = Git.open '.'
+      a = instance.status.added.keys # Added files
+      u = instance.status.untracked.keys # Untracked files
+      c = instance.status.changed.keys # Changed files
+      p = path
+      p.remove(Dir.pwd.to_s+'/'.to_s)
+      if p==path
+        p=''
+      else
+        p+='/'
+      end
+      return 'A' if a.any? { |x| x.include? "#{p}#{content}" }
+      return 'U' if u.any? { |x| x.include? "#{p}#{content}" }
+      return 'C' if c.any? { |x| x.include? "#{p}#{content}" }
+      p = '-'
+    end
```

Comment:
Maybe this would seem more readable and also solve the rubocop complexity issue?

```ruby
def git_info(path, content)
   until File.exist?('.git') # check whether the repository is git controlled
     return '' if Dir.pwd=='/'
     Dir.chdir('..')
   end

   relative_path = path.remove(Dir.pwd+'/')
   relative_path = relative_path==path ? '' : relative_path+'/'

   status = Git.open('.').status
   return 'A' if status.added.keys.any? { |a| a.include?("#{relative_path}#{content}") }
   return 'U' if status.untracked.keys.any? { |u| u.include?("#{relative_path}#{content}") }
   return 'C' if status.changed.keys.any? { |c| c.include?("#{relative_path}#{content}") }
   '-'
end
```